dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,626 @@
+/*
+    DSSCollection contains collection group & completion functionality for DSS.
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import NonFungibleToken from "./NonFungibleToken.cdc"
+import MetadataViews from 0xMETADATAVIEWSADDRESS
+
+// The DSSCollection contract
+//
+pub contract DSSCollection: NonFungibleToken {
+
+    // Contract Events
+    //
+    pub event ContractInitialized()
+
+    // NFT Collection Events
+    //
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+
+    // Events
+    //
+    pub event CollectionGroupCreated(
+        id: UInt64,
+        name: String,
+        typeName: String,
+        startTime: UFix64?,
+        endTime: UFix64?,
+        timeBound: Bool
+    )
+    pub event CollectionGroupClosed(id: UInt64)
+    pub event SlotCreated(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        logicalOperator: String,
```

Comment:
what is the logical operator for? does it tie slots together or items under a slot?

Reply:
It ties all the items in the slot together. An example of OR:

collect:
Lebron James S1 OR Kyrie Irving S1 OR Kevin Durant S2

An example of AND:

collect:
Lebron James S1 AND Lebron James S2 AND Lebron James S3 AND Lebron James S4

Reply:
i see, tho, i don't think the AND is useful in a slot, since a slot can only take 1 item, you can't have 1 item that fulfills all those conditions

Reply:
A slot can have many items, it's an array. If a slot has more than one Item and the logical operator is "AND", you would need to collect all the items in the slot to fulfill the slot & get the completion token.

Reply:
oh, i see. That makes sense